lib/model/model.go

```diff
@@ -1147,6 +1147,10 @@ type clusterConfigDeviceInfo struct {
 	local, remote protocol.Device
 }
 
+type ClusterConfigReceivedEventData struct {
+	ID string `json:"id"`
```

Comment:
Same here: I'd leave the `protocol.DeviceID` type here and leave conversion to string to marshalling (already happening by default through `.MarshalText`).

Reply:
Actually I just noticed that the custom event data type doesn't work.  It panics when `processUpdate()` in the `folderSummaryService` tries to typecast it to `map[string]string`.

Reply:
You'll have to handle it in it's own switch-case-branch, casting to `ClusterConfigReceivedEventData`. That reminds me: The new event also needs to be added to the verbose-service (lib/syncthing).

Reply:
Isn't this all pretty special-case?  I mean almost all the other code uses simple maps, why start defining custom types on this one event if it brings so much additional complexity?  The rest of the code seems to assume that any event data can be casted to `map[string]string`.

Reply:
There's a mix of `map[string]string` and `map[string]interface{}`. And mostly that's not used at all except for json/logging. I recently needed to use some event data for actual logic, and that's when things get annoying if you need to do all the type assertions and having a single, defined struct comes in handy. That's why I proposed using those going forward. If you don't want to add the complexity (don't think it's that complex though), going with `map[string]interface{}` seems like the sensible option, as we already know that we might be adding more data (from the CC) to this event in the future.

Reply:
I think I will go for `map[string]string` for now.  Even using `map[string]interface{}` has a high potential to panic, because the cast to `map[string]string` is allowed by the compiler but fails at runtime.  That's too high a risk for me to start with a half-baked (although better long-term) solution.

Right now I think we need to fix this ASAP in the most future-compatible way or actually revert the original addition in order not to delay the RC due today any longer.